tools/tcat_ble_client/cli/base_commands.py

```diff
@@ -54,55 +55,112 @@ async def execute_default(self, args, context):
         return CommandResultNone()
 
 
-class HelloCommand(Command):
+class BleCommand(Command):
 
-    def get_help_string(self) -> str:
-        return 'Send round trip "Hello world!" message.'
+    @abstractmethod
+    def get_log_string(self) -> str:
+        pass
+
+    @abstractmethod
+    def prepare_data(self, context):
+        pass
 
     async def execute_default(self, args, context):
         bless: BleStreamSecure = context['ble_sstream']
-        print('Sending hello world...')
-        data = TLV(TcatTLVType.APPLICATION.value, bytes('Hello world!', 'ascii')).to_bytes()
+
+        print(self.get_log_string())
+        data = self.prepare_data(context)
         response = await bless.send_with_resp(data)
         if not response:
             return
         tlv_response = TLV.from_bytes(response)
         return CommandResultTLV(tlv_response)
 
 
-class CommissionCommand(Command):
+class HelloCommand(BleCommand):
```

Comment:
Now that "ping" support is added, the Hello command could just use the Ping TLV underneath. The TCAT device then sends back "Hello World!" as echo response.

Reply:
@EskoDijk This might be unfortunate but this functionality is there to demonstrate not echo requests but rather handling of user communication channel. So I would leave it as is.

Reply:
Ok, this is different from ping then. Later on we can update this command to follow the spec on this: first select an application layer using 0x80, and then sending the data.